transfers/transfer_async.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Automate Transfers.
+
+Helper script to automate running transfers through Archivematica.
+
+Similar to ``transfers.transfer`` but using the new `/api/v2beta` API when
+possible.
+"""
+
+from __future__ import print_function, unicode_literals
+
+import base64
+import os
+import sys
+
+import requests
+
+# Allow execution as an executable and the script to be run at package level
+# by ensuring that it can see itself.
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+from transfers import transfer
+from transfers.loggingconfig import set_log_level
+from transfers.models import Unit
+from transfers.transferargs import get_parser
+from transfers.transfer import (
+    LOGGER, get_next_transfer, get_accession_id, main
+)
+from transfers.utils import fsencode
+
+
+class DashboardAPIError(Exception):
+    """Dashboard API error."""
+
+
+def _api_create_package(am_url, am_user, am_api_key,
+                        name, type, accession,
+                        ts_location_uuid, ts_path):
+    url = am_url + '/api/v2beta/package/'
+    headers = {
+        'Authorization': 'ApiKey {}:{}'.format(am_user, am_api_key)
+    }
+    data = {
+        'name': name,
+        'type': type,
+        'accession': accession,
+        'path': base64.b64encode(fsencode(ts_location_uuid) + b':' + ts_path),
+    }
+    LOGGER.debug('URL: %s; Headers: %s, Data: %s', url, headers, data)
+    response = requests.post(url, headers=headers, json=data)
+    response.raise_for_status()
+    LOGGER.debug('Response: %s', response)
+    resp_json = response.json()
+    error = resp_json.get('error')
+    if error:
+        raise DashboardAPIError(error)
+    return resp_json
```

Comment:
I would consider adding functionality from here to the `utils.py` script where applicable. This is very rough code here as I work out how to replace the current call, but I've been finding setting the headers for authentication more reliable (I don't think auth in a JSON body is supported by all our endpoints?) https://github.com/artefactual/automation-tools/blob/dev/client-reingest-compressed/transfers/utils.py#L64

But, my original issue https://github.com/artefactual/automation-tools/issues/52 may also have handled what you're doing here, grabbing the response details and being able to access them possibly via member vars, and not just having to rely on logging inside `utils.py`.

Reply:
I agree. Right now this is just doing things in the same way as it's done in `transfers.transfer.start_transfer` but I think we can get to your idea soon.

Reply:
I see, we're just trying to get to the point that `transfers.py` is at first and build from there. Works for me.